### DISCUSSION TITLE
Rename "repository" to "deck"

### DISCUSSION THREAD

**Discussion by kklecho** [dagster-io/dagster#6301] [Ideas]
Using another term for collection of jobs might reduce confusion for newcomers and add efficiency to communication.

Hi all, my first post here, thanks for your great work on dagster, it's great.

**Comment by yuhan** [comment]
Thanks for the feedback @kklecho. This is something we are actively thinking of now!

Could you please elaborate on why "deck" and which part(s) of "repository" are causing the biggest confusion?

cc @dpeng817 regarding repository abstraction and the potential renaming.

**Reply by dpeng817** [reply]
following this discussion; we're currently considering substantial re-works of the repository abstraction, including a re-name.

**Reply by kklecho** [reply]
As a newcomer, I see this:
![image](https://user-images.githubusercontent.com/3323741/172591308-f3734c00-666d-4e90-a193-dcee6e989a95.png)
and I think I'll have a separate git repo for collection of jobs. The meaning of repository got so tied to the VCS that people get confused when it's not. Eg. when people google "dagster repository" they are probably thinking dagster github. 

As to what name instead - I like "dagster deck". The "deck" is not as common. And less abstract than "collection". And is separate name for separate concept. 

"Job collection" would be more clear, but sounds a bit clunky. And is untrue, that's more than just collection of jobs. And the name contains "job" - that mixes in the other core concept. IICS has this concept "task flow" for the collection of tasks, I don't like that name for the same reason.

When you think in visual terms and the job's graph fits on a card, it would be natural to think of the deck as extension.

**Reply by yuhan** [reply]
Hi @kklecho thanks again for the feedback! We've come up with a RFC [here](https://github.com/dagster-io/dagster/discussions/10772) to eliminate repository aiming to reduce the confusion you described.
